src/db/write.rs

```diff
@@ -106,16 +116,28 @@ impl<'a> Write<'a> {
     }
 
     pub fn put(&mut self, key: Vec<u8>, val: Vec<u8>) {
+        match self.map.get(&key) {
```

Comment:
[minifix] Internally map knows if the put() replaced an item (because its backing store returns that information from put). Change map.put() to return that to avoid the read here.